documenting jupyter magics in docs (#240)
ploomber/jupysql: doc/quick-start.md

    extension: .md
    format_name: myst
    format_version: 0.13
    jupytext_version: 1.14.4
    jupytext_version: 1.14.5
kernelspec:
  display_name: Python 3 (ipykernel)
  language: python
  name: python3

> myst:

Reviewer: The metadata disappeared for this file.

I recently discovered this problem, jupyter will delete it once we save the file. the solution is to copy it back from the master branch, and add `notebook_metadata_filter: myst`, see the example at the   bottom of this page: https://github.com/ploomber/contributing/blob/main/documentation/notebooks.md

Author: @edublancas I copy and paste the whole part from the link you provided. Not sure if it is expected to be done in this way. Is there any specific command you want to add on this file?

Reviewer 2: So I guess Eduardo means:
1. Make [this part](https://github.com/ploomber/jupysql/blob/master/doc/quick-start.md?plain=1#L1-L19) same as what we had in master branch
2. Add `notebook_metadata_filter: myst` inside

Also if `notebook_metadata_filter: myst` is required to add in each `.md` file, we can have another PR to handle other `.md` files

Reviewer 2: (I am not sure if 
```
  cell_metadata_filter: -all
  formats: md:myst
```
should appear in this file

  display_name: Python 3 (ipykernel)
  language: python
  name: python3
myst:
  html_meta:
    description lang=en: "Quickstart for JupySQL: a package to run SQL in Jupyter"
    keywords: jupyter, sql, jupysql

>     property=og:locale: en_US

Reviewer: there are still problems here, you need to copy back the html_meta section into your file: https://raw.githubusercontent.com/ploomber/jupysql/master/doc/quick-start.md and push again

```{code-cell} ipython3
df.head()
```

# Jupyter Magic

> One may be unfamiliar with the commands prefixed with '%' used in this instruction. Here is a detailed description of this command and its usage. 

Reviewer 2: ```suggestion
One may be unfamiliar with the commands prefixed with `%` used in this instruction. Here is a detailed description of this command and its usage. 
```

## Definition of Jupyter Magic

Magics are specific to and provided by the IPython kernel. Some common usage of magic functions are: running external code files, timing code execution, and loading IPython Extensions. 

Suppose execute.py is a python code file

```{tip}
%run execute.py

> %timeit L = [n ** 2 for n in range(1000)] (Timeing executions -- will return 1000 loops, best of 3: 325 µs per loop)

Reviewer 2: Typo: `Timeing` -> `Timing`

```

> In our code above,  we use **%load_ext** to load an IPython extension by its module name, 'sql', and then directly use the extension using '%sql'.

Reviewer 2: ``
```suggestion
In our code above, we use **%load_ext** to load an IPython extension by its module name, `sql`, and then directly use the extension by using `%sql`.
```

```{tip}
load an IPython extension by its module name.
%load_ext sql 
```

## Line Magic VS Cell Magic

**Line magics**, which are denoted by a single % prefix and operate on a single line of input, and **cell magics**, which are denoted by a double %% prefix and operate on multiple lines of input. 

For example, for the code above, **%sql** is a line magic, and **%%sql** is a code magic. 

### Reference 

> IPython doc: https://ipython.readthedocs.io/en/stable/interactive/magics.html#cell-magics 

Reviewer 2: ```suggestion
[IPython doc](https://ipython.readthedocs.io/en/stable/interactive/magics.html#cell-magics)
```

> Python Data Science Handbook: https://jakevdp.github.io/PythonDataScienceHandbook/01.03-magic-commands.html

Reviewer 2: ```suggestion
[Python Data Science Handbook](https://jakevdp.github.io/PythonDataScienceHandbook/01.03-magic-commands.html)
```

```{code-cell} ipython3
df.head()
```

# Jupyter Magic

One may be unfamiliar with the commands prefixed with '%' used in this instruction. Here is a detailed description of this command and its usage. 

## Definition of Jupyter Magic

Magics are specific to and provided by the IPython kernel. Some common usage of magic functions are: running external code files, timing code execution, and loading IPython Extensions. 

```{code-cell} ipython3

# Suppose execute.py is a python code file

> %run execute.py

Reviewer 2: This is the main reason why it builds fail, since every code inside `{code-cell} ipython3` section will actual run and look up for `execute.py` file to execute

If you just want to explain the usage then you can just move it out this code-cell section & use regular markdown format to explain

Reviewer: @chuwpeng123 please address this, you need to convert this to markdown cells - this will prevent our doc building process for running them. Also, I'd suggest moving this to a new file called FAQ and put it at the end of the user guide, then link from the quickstart to the FAQ document via an [admonition](https://jupyterbook.org/en/stable/reference/cheatsheet.html#admonitions)

more info here: https://github.com/ploomber/contributing/blob/main/documentation/notebooks.md

Author: @edublancas I am looking at it. Try to get this down by tomorrow.
